Firestore/Source/Local/FSTLocalStore.mm

```diff
@@ -448,20 +448,23 @@ - (FSTDocumentDictionary *)executeQuery:(FSTQuery *)query {
 }
 
 - (FSTDocumentKeySet *)remoteDocumentKeysForTarget:(FSTTargetID)targetID {
-  return [self.queryCache matchingKeysForTargetID:targetID];
+  FSTWriteGroup *group = [self.persistence startGroupWithAction:@"RemoteDocumentKeysForTarget"];
```

Comment:
As above, I would prefer to do the switch to transaction blocks all in one go, after having identified all the places they need to exist. Otherwise, we end up intermingling them where you can use a transaction block as long as everything in the transaction is a read. Otherwise, you need a group anyways for anything that's not ported yet, and then it won't work in a transaction block due to the one-transaction-at-a-time rule.